### PR TITLE
Fixes #123 - improve browsing params handling

### LINKED ISSUE #123: Improve Browsing UX
@rhyslbw identified a few issues with the browsing feature:

## Params validation
There's a bug with the browse-* pages when the total of items is less than the page qty (eg epochs in the dev data set). It triggers an endless loop:

## Drop redirect because it breaks browser navigation
Need to drop the redirect to the "correct" url and just infer the corrected details from the given prams (this is already possible with the current logic)

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR fixes a couple of issues with the browsing implementation:

- Handle invalid `upper` params correctly
- No redirect on invalid / incomplete params (just interpret best as possible), results in proper browser navigation support.
- Changes the default number of epochs and blocks to 20 and increase max to 50.

### CODE REVIEW

File: source/widgets/browsing/BrowseInRange.tsx [hunk 1]
```diff
@@ -93,12 +93,8 @@ export const BrowseInRange = ({
     bounds.upper.toString() === userParamUpper;
```
Comment (rhyslbw) [inline source/widgets/browsing/BrowseInRange.tsx]: `isCorrectPath` is now unused

Comment (rhyslbw) [inline source/widgets/browsing/BrowseInRange.tsx]: `isCorrectPath` is now unused

**Comment by DominikGuzei** [comment]
@rhyslbw yeah it's 21 blocks but wouldn't you expect to see all blocks in the given range? The params say `31070 -> 31050` so it seems legit to display all of these, right?

**Comment by rhyslbw** [comment]
> @rhyslbw yeah it's 21 blocks but wouldn't you expect to see all blocks in the given range? The params say `31070 -> 31050` so it seems legit to display all of these, right?

With `BLOCKS_PER_PAGE_DEFAULT = 20` I'd expect to see 20 blocks on the page, not 21

File: source/features/blocks/ui/BlocksBrowser.tsx [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { useNetworkInfoFeature } from '../../network-info/context';
 import { useBlocksFeature } from '../context';
 import BlockList from './BlockList';
 
+// TODO: This actually fetches 1 item more than given
```
Comment (rhyslbw) [inline source/features/blocks/ui/BlocksBrowser.tsx]: 👍

**Pull request merged**